windrose/windrose.py

```diff
@@ -376,6 +376,11 @@ def _init_plot(self, direction, var, **kwargs):
         bins = np.asarray(bins)
         nbins = len(bins)
 
+        if np.isnan(bins).any():
+            raise ValueError(
+                "Input bins contain NaN values. If bins are not explicitly "\
+                "provided as 1D array, that means var array has NaN values.")
```

Comment:
You don't need the `\` for multiple lines inside `()`.

```suggestion
                "Input bins contain NaN values. If bins are not explicitly "
                "provided as 1D array, that means var array has NaN values.")
```

PS: The message is a bit verbose. I don't have a strong opinion but how about:

"Could not compute the bins due to the presence of NaNs in either the bins provided or the original data."

?